src/shared/components/CustomButton/utils/FontDetector.ts

```diff
@@ -0,0 +1,89 @@
+ *          to parent element if the font is missing.
+ * Version: 0.2 (04 Mar 2012)
+ *          Comparing font against all the 3 generic font families ie,
+ *          'monospace', 'sans-serif' and 'sans'. If it doesn't match all 3
+ *          then that font is 100% not available in the system
+ * Version: 0.3 (24 Mar 2012)
+ *          Replaced sans with serif in the list of baseFonts
+ * TypeScript Reactor: July 3, 2024
+ */
+
+/**
+ * Usage: d = new Detector();
+ *        d.detect('font name');
+ */
+
+export interface IFontDetector {
+    detect: (font: string) => boolean;
+}
+
+export class FontDetector implements IFontDetector {
+    // a font will be compared against all the three default fonts.
+    // and if it doesn't match all 3 then that font is not available.
+    baseFonts = ['monospace', 'sans-serif', 'serif'];
+
+    // we use m or w because these two characters take up the maximum width.
+    // And we use a LLi so that the same matching fonts can get separated
+    testString = 'mmmmmmmmmmlli';
+
+    // we test using 72px font size, we may use any size. I guess larger the better.
+    testSize = '72px';
+
+    detect: (font: string) => boolean;
+
+    constructor() {
+        // precompute for the test
+        var defaultWidth: { [key: string]: number } = {};
+        var defaultHeight: { [key: string]: number } = {};
+
+        var html = document.getElementsByTagName('body')[0];
+
+        // create a SPAN in the document to get the width of the text we use to test
+        var span = document.createElement('span');
+        span.style.fontSize = this.testSize;
+        span.innerHTML = this.testString;
+
+        const baseFonts = this.baseFonts;
+        for (var index in baseFonts) {
+            //get the default width for the three base fonts
+            span.style.fontFamily = baseFonts[index];
+            html.appendChild(span);
+            defaultWidth[baseFonts[index]] = span.offsetWidth;
+            defaultHeight[baseFonts[index]] = span.offsetHeight;
+            html.removeChild(span);
+        }
+
+        // expose a detect() function that leverages that state
+        this.detect = (font: string): boolean => {
+            // console.log("detect:" + font);
+            for (var index in baseFonts) {
+                // name of the font along with the base font for fallback.
+                span.style.fontFamily = font + ',' + baseFonts[index];
+                // add the span with the test font, and see if it's actually using a baseFont
+                html.appendChild(span);
+                var matched =
+                    span.offsetWidth != defaultWidth[baseFonts[index]] ||
+                    span.offsetHeight != defaultHeight[baseFonts[index]];
+                html.removeChild(span);
+                if (matched) {
+                    return true;
+                }
+            }
+            return false;
+        };
+    }
+}
```

Comment:
This utility is a workaround to basically check a boolean "installed" flag set on the client system. It appears there isn't a way to check if a particular URL protocol is supported, or a particular software is installed, so this was the workaround that I found. The software installs a special font, then the frontend code can check if that font is installed by indirectly checking if a fallback font is used (by looking at sizes).

This code is modified from the original to be more efficient.

ALTERNATIVES
A) We could remove the FontDetector code, then the CustomButton would appear depending on UserAgent only (e.g. all Windows browsers). That could work, since I believe we can detect if launching the URL protocol failed then direct the user to another page.  
B) If there is another way to detect if a particular URL protocol is supported or a particular software is installed, then we could try that. However, my research into the issue seemed to conclude there wasn't.